api/webhook.go

```diff
@@ -484,27 +452,6 @@ func PostWebhook(c *gin.Context) {
 			config = pipeline.GetData()
 		}
 
-		// send API call to capture repo for the counter
-		r, err = database.FromContext(c).GetRepoForOrg(r.GetOrg(), r.GetName())
-		if err != nil {
-			retErr := fmt.Errorf("%s: unable to get repo %s: %w", baseErr, r.GetFullName(), err)
-
-			// check if the retry limit has been exceeded
-			if i < retryLimit-1 {
-				logrus.WithError(retErr).Warningf("retrying #%d", i+1)
-
-				// continue to the next iteration of the loop
-				continue
-			}
-
-			util.HandleError(c, http.StatusBadRequest, retErr)
-
-			h.SetStatus(constants.StatusFailure)
-			h.SetError(retErr.Error())
-
-			return
-		}
```

Comment:
This is the duplicate call to `GetRepoForOrg` that I mentioned in the PR description. The removal of this entire block resulted in zero changes based on my local testing. To my understanding, this is called to adjust the build counter, but there are no changes to the repo between this point and the first time we call `GetRepoForOrg`, let alone any calls to update the repo object in the database.

Reply:
if i'm not mistaken, this part was to address some race conditions with simultaneous webhooks coming in for the same repo. one might increment the counter. the other call will get the same repo (and counter at the time) but get blocked with a duplicate db error. hence, we try to get the updated counter. this code is a little sensitive and changes probably should be tested, especially with regards to concurrency (unless you did that?).

Reply:
I'll just remove this then since it's not related to the PR, and I'm not really in the mood to test out concurrency when all I want is repo topics.

I would have liked to see a comment on why this DB call is needed.

Reply:
agreed. my initial version of the comment did point out that it would be nice to have more commentary, especially in this vicinity. the comment at the top of the retry loop kind of touches on it, but probably could be better.